packages/one-app-runner/src/startApp.js

```diff
@@ -124,24 +162,75 @@ module.exports = async function startApp({
-  const appPort = process.env.HTTP_PORT || 3000;
-  const devCDNPort = process.env.HTTP_ONE_APP_DEV_CDN_PORT || 3001;
-  const devProxyServerPort = process.env.HTTP_ONE_APP_DEV_PROXY_SERVER_PORT || 3002;
-  const metricsPort = process.env.HTTP_METRICS_PORT || 3005;
-  const debugPort = process.env.HTTP_ONE_APP_DEBUG_PORT || 9229;
-  const ports = `-p ${appPort}:${appPort} -p ${devCDNPort}:${devCDNPort} -p ${devProxyServerPort}:${devProxyServerPort} -p ${metricsPort}:${metricsPort} -p ${debugPort}:${debugPort}`;
-
-  const command = `${generatePullCommand()} docker run -t ${ports} -e NODE_ENV=development ${generateContainerNameFlag()} ${generateNetworkToJoin()} ${generateEnvironmentVariableArgs(envVars)} ${generateModuleMountsArgs(modulesToServe)} ${generateCaCertsCommands(envVars)} ${appDockerImage} /bin/sh -c "${generateServeModuleCommands(modulesToServe)} ${generateSetMiddlewareCommand(parrotMiddlewareFile)} ${generateSetDevEndpointsCommand(devEndpointsFile)} node ${generateDebug(debugPort)} lib/server/index.js --root-module-name=${rootModuleName} ${generateModuleMap()} ${generateUseMocksFlag(parrotMiddlewareFile)} ${generateUseHostFlag()}"`;
-  const dockerProcess = spawn(command, { shell: true });
-  dockerProcess.on('error', () => {
+
+  const appPort = Number.parseInt(process.env.HTTP_PORT, 10) || 3000;
+  const devCDNPort = Number.parseInt(process.env.HTTP_ONE_APP_DEV_CDN_PORT, 10) || 3001;
+  const devProxyServerPort = Number.parseInt(
+    process.env.HTTP_ONE_APP_DEV_PROXY_SERVER_PORT, 10
+  ) || 3002;
+  const metricsPort = Number.parseInt(process.env.HTTP_METRICS_PORT, 10) || 3005;
+  const debugPort = Number.parseInt(process.env.HTTP_ONE_APP_DEBUG_PORT, 10) || 9229;
+
+  const ports = [
+    appPort,
+    devCDNPort,
+    devProxyServerPort,
+    metricsPort,
+    debugPort,
+  ];
+
+  const containerEnvVars = generateEnvironmentVariableArgs(envVars);
+
+  const mounts = new Map([
+    ...(modulesToServe || []).map((currentValue) => [currentValue, `/opt/module-workspace/${path.basename(currentValue)}`]),
+  ]);
+
+  const hostNodeExtraCaCerts = envVars.NODE_EXTRA_CA_CERTS || process.env.NODE_EXTRA_CA_CERTS;
+  if (hostNodeExtraCaCerts) {
+    console.log('mounting host NODE_EXTRA_CA_CERTS');
+    const mountPath = '/opt/certs.pem';
+    mounts.set(hostNodeExtraCaCerts, mountPath);
+    containerEnvVars.set('NODE_EXTRA_CA_CERTS', mountPath);
+  }
+
+  const containerShellCommand = `${
+    generateServeModuleCommands(modulesToServe)
+  } ${
+    generateSetMiddlewareCommand(parrotMiddlewareFile)
+  } ${
+    generateSetDevEndpointsCommand(devEndpointsFile)
+  } node ${
+    generateDebug(debugPort, useDebug)
+  } lib/server/index.js --root-module-name=${rootModuleName} ${
+    generateModuleMap(moduleMapUrl)
+  } ${
+    generateUseMocksFlag(parrotMiddlewareFile)
+  } ${
+    generateUseHostFlag()
+  }`;
+
+  try {
+    if (!offline) {
+      await dockerPull(appDockerImage);
```

Comment:
can we use https://www.npmjs.com/package/dockerode instead? 🤔 

could also
```javascript
async function dockerCreateNetwork(name) {
  return new Promise((resolve, reject) => {
    const pullProcess = spawn('docker', ['network', 'create', name]);
    pullProcess.stdout.pipe(process.stdout);
    pullProcess.stderr.pipe(process.stderr);
    pullProcess.on('close', (code) => {
      if (code !== 0) {
        return reject(code);
      }
      return resolve(code);
    });
  });
}
```
and remove the dockerode dependency?

Reply:
lines 20-78 of the diff are relevant context

Reply:
I think we could remove dockerode usage, but that's probably best in a different PR